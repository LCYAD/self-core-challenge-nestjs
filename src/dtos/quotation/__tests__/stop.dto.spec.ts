import {
  generateInvalidStringFormatTest,
  generateMissingFieldTest,
  generateValidBaseTest,
  generateValidOptionalTest
} from '@tests/helper/dto.helper'

import {
  QuotationStopCoordinatesDto,
  quotationStopCoordinatesValidExample,
  QuotationStopDto,
  quotationStopValidExample
} from '../stop.dto'

describe('QuotationStopCoordinatesDto', () => {
  describe('valid response', () => {
    generateValidBaseTest(
      QuotationStopCoordinatesDto,
      quotationStopCoordinatesValidExample
    )
  })

  describe('invalid response', () => {
    generateMissingFieldTest(
      ['lat', 'lng'],
      QuotationStopCoordinatesDto,
      quotationStopCoordinatesValidExample
    )
    generateInvalidStringFormatTest(
      [
        { key: 'lat', invalidString: 'abc' },
        { key: 'lng', invalidString: 'abc' }
      ],
      QuotationStopCoordinatesDto,
      quotationStopCoordinatesValidExample
    )
  })
})

describe('QuotationStopDto', () => {
  describe('valid response', () => {
    generateValidBaseTest(QuotationStopDto, quotationStopValidExample)
    generateValidOptionalTest(
      ['coordinates'],
      QuotationStopDto,
      quotationStopValidExample
    )
  })

  describe('invalid response', () => {
    generateMissingFieldTest(
      ['address'],
      QuotationStopDto,
      quotationStopValidExample
    )
  })
})
