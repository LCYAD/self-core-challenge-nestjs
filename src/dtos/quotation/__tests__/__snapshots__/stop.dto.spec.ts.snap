// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuotationStopCoordinatesDto invalid response for lat not a allowed value abc 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "matches": "lat must match /^(\\+|-)?(?:90(?:(?:\\.0{1,6})?)|(?:[0-9]|[1-8][0-9])(?:(?:\\.[0-9]{1,6})?))$/ regular expression",
    },
    "property": "lat",
    "value": "abc",
  },
]
`;

exports[`QuotationStopCoordinatesDto invalid response for lng not a allowed value abc 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "matches": "lng must match /^(\\+|-)?(?:180(?:(?:\\.0{1,6})?)|(?:[0-9]|[1-9][0-9]|1[0-7][0-9])(?:(?:\\.[0-9]{1,6})?))$/ regular expression",
    },
    "property": "lng",
    "value": "abc",
  },
]
`;

exports[`QuotationStopCoordinatesDto invalid response for missing lat 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "lat must be a string",
      "matches": "lat must match /^(\\+|-)?(?:90(?:(?:\\.0{1,6})?)|(?:[0-9]|[1-8][0-9])(?:(?:\\.[0-9]{1,6})?))$/ regular expression",
    },
    "property": "lat",
    "value": undefined,
  },
]
`;

exports[`QuotationStopCoordinatesDto invalid response for missing lng 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "lng must be a string",
      "matches": "lng must match /^(\\+|-)?(?:180(?:(?:\\.0{1,6})?)|(?:[0-9]|[1-9][0-9]|1[0-7][0-9])(?:(?:\\.[0-9]{1,6})?))$/ regular expression",
    },
    "property": "lng",
    "value": undefined,
  },
]
`;

exports[`QuotationStopDto invalid response for missing address 1`] = `
[
  ValidationError {
    "children": [],
    "constraints": {
      "isString": "address must be a string",
    },
    "property": "address",
    "value": undefined,
  },
]
`;
